File: examples/llmix/typescript/quickstart.ts
/**
 * LLMix TypeScript quickstart.
 *
 * Run with:
 *   OPENAI_API_KEY=sk-... bun run examples/llmix/typescript/quickstart.ts
 */

import {
  CallPipeline,
  KeyPool,
  TwoTierCache,
  openaiDispatch,
} from "@snoai/llmix";

async function main(): Promise<void> {
  const pipeline = new CallPipeline({
    dispatch: openaiDispatch(),
    responseCache: new TwoTierCache("memory"),
  });
  pipeline.setKeyPool("openai", new KeyPool([process.env["OPENAI_API_KEY"]!]));

  const response = await pipeline.call({
    config: {
      provider: "openai",
      model: "gpt-4o-mini",
      common: { temperature: 0.7, maxOutputTokens: 256 },
      caching: { strategy: "memory" },
    },
    messages: [
      { role: "user", content: "In one sentence, what is LLMix?" },
    ],
  });

  console.log(response.content);
  console.log(`cache_hit=${response.cacheHit} usage=${JSON.stringify(response.usage)}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/llmix/python/quickstart.py
"""LLMix Python quickstart.

Run with:
    OPENAI_API_KEY=sk-... uv run --project packages/llmix/python python examples/llmix/python/quickstart.py
"""

import asyncio
import os

from llmix import (
    CallInput,
    CallPipeline,
    KeyPool,
    PipelineConfig,
    TwoTierCache,
    openai_dispatch,
)


async def main() -> None:
    pipeline = CallPipeline(
        PipelineConfig(
            dispatch=openai_dispatch(),
            response_cache=TwoTierCache("memory"),
        )
    )
    pipeline.set_key_pool("openai", KeyPool([os.environ["OPENAI_API_KEY"]]))

    response = await pipeline.call(
        CallInput(
            config={
                "provider": "openai",
                "model": "gpt-4o-mini",
                "common": {"temperature": 0.7, "max_output_tokens": 256},
                "caching": {"strategy": "memory"},
            },
            messages=[
                {"role": "user", "content": "In one sentence, what is LLMix?"}
            ],
        )
    )

    print(response.content)
    print(f"cache_hit={response.cache_hit} usage={response.usage}")


if __name__ == "__main__":
    asyncio.run(main())
